test/functional/wallet_migration.py

```diff
@@ -0,0 +1,405 @@
+
+    def test_multisig(self):
+        default = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+
+        # Contrived case where all the multisig keys are in a single wallet
+        self.log.info("Test migration of a wallet with all keys for a multisig")
+        multisig0 = self.create_legacy_wallet("multisig0")
+        addr1 = multisig0.getnewaddress()
+        addr2 = multisig0.getnewaddress()
+        addr3 = multisig0.getnewaddress()
+
+        ms_info = multisig0.addmultisigaddress(2, [addr1, addr2, addr3])
+
+        multisig0.migratewallet()
+        assert_equal(multisig0.getwalletinfo()["descriptors"], True)
+        self.assert_is_sqlite("multisig0")
+        ms_addr_info = multisig0.getaddressinfo(ms_info["address"])
+        assert_equal(ms_addr_info["ismine"], True)
+        assert_equal(ms_addr_info["desc"], ms_info["descriptor"])
+        assert_equal("multisig0_watchonly" in self.nodes[0].listwallets(), False)
+        assert_equal("multisig0_solvables" in self.nodes[0].listwallets(), False)
+
+        pub1 = multisig0.getaddressinfo(addr1)["pubkey"]
+        pub2 = multisig0.getaddressinfo(addr2)["pubkey"]
+
+        # Some keys in multisig do not belong to this wallet
+        self.log.info("Test migration of a wallet that has some keys in a multisig")
+        self.nodes[0].createwallet(wallet_name="multisig1")
+        multisig1 = self.nodes[0].get_wallet_rpc("multisig1")
+        ms_info = multisig1.addmultisigaddress(2, [multisig1.getnewaddress(), pub1, pub2])
+        ms_info2 = multisig1.addmultisigaddress(2, [multisig1.getnewaddress(), pub1, pub2])
+        assert_equal(multisig1.getwalletinfo()["descriptors"], False)
+
+        addr = ms_info["address"]
+        txid = default.sendtoaddress(addr, 10)
+        multisig1.importaddress(addr)
+        assert_equal(multisig1.getaddressinfo(ms_info["address"])["ismine"], False)
+        assert_equal(multisig1.getaddressinfo(ms_info["address"])["iswatchonly"], True)
+        assert_equal(multisig1.getaddressinfo(ms_info["address"])["solvable"], True)
+        self.generate(self.nodes[0], 1)
+        multisig1.gettransaction(txid)
+        assert_equal(multisig1.getbalances()["watchonly"]["trusted"], 10)
+        assert_equal(multisig1.getaddressinfo(ms_info2["address"])["ismine"], False)
+        assert_equal(multisig1.getaddressinfo(ms_info2["address"])["iswatchonly"], False)
+        assert_equal(multisig1.getaddressinfo(ms_info2["address"])["solvable"], True)
+
+        # Migrating multisig1 should see the multisig is no longer part of multisig1
+        # A new wallet multisig1_watchonly is created which has the multisig address
+        # Transaction to multisig is in multisig1_watchonly and not multisig1
+        multisig1.migratewallet()
+        assert_equal(multisig1.getwalletinfo()["descriptors"], True)
+        self.assert_is_sqlite("multisig1")
+        assert_equal(multisig1.getaddressinfo(ms_info["address"])["ismine"], False)
```

Comment:
fb5e0957e066bd0825947a399242242f1f1a2ae6 : Maybe use `addr` instead of `ms_info["address"]`

Reply:
Done